x-pack/plugins/gis/public/shared/layers/vector_layer.js

```diff
@@ -269,34 +278,44 @@ export class VectorLayer extends ALayer {
   }
 
 
+  _getSearchFilters(dataFilters) {
+    const fieldNames = [
+      ...this._source.getFieldNames(),
+      ...this._style.getSourceFieldNames(),
+      ...this.getValidJoins().map(join => {
+        return join.getLeftFieldName();
+      })
+    ];
+
+    const targetPrecision = getGeohashPrecisionForZoom(dataFilters.zoom);
+    return {
+      ...dataFilters,
+      fieldNames: _.uniq(fieldNames).sort(),
+      geohashPrecision: targetPrecision
+    };
+  }
+
   async _syncSource({ startLoading, stopLoading, onLoadError, dataFilters }) {
+
     const sourceDataId = 'source';
     const requestToken = Symbol(`layer-source-refresh:${ this.getId()} - source`);
-    try {
```

Comment:
this try-catch seemed to be unnecessarily wide. this encloses the data-request code a little more tightly